### ISSUE TITLE
`Attempted to construct an Id from a null pointer`

### ISSUE THREAD

**Issue by 42plamusse** [balthild/native-dialog-rs#25]
System: MacOS

```
   0: std::panicking::begin_panic
   1: native_dialog::dialog_impl::mac::ffi::cocoa::url::INSURL::from_str
   2: native_dialog::dialog_impl::mac::file::<impl native_dialog::dialog::DialogImpl for native_dialog::dialog::file::SaveSingleFile>::show
   3: native_dialog::file::FileDialog::show_save_single_file
```

In `native_dialog::dialog_impl::mac::ffi::cocoa::url::INSURL::from_str`, `Id::from_ptr` receives a `0x0` pointer.

**Comment by 42plamusse** [comment]
This occurs when the path contains whitespaces (folder `Application Support`)

**Comment by 42plamusse** [comment]
Replacing whitespace by `%20` fixes it.

```rust
pub fn game_root_config_path() -> String {
    let path = sdl2::filesystem::pref_path(crate::constant::ORG_NAME, crate::constant::APP_NAME)
        .expect("a prefered config");
    if cfg!(target_os = "macos") {
        path.replace(' ', "%20")
    } else {
        path
    }
}
```

This could be an useful crate to fix it globally https://docs.rs/urlencoding/latest/urlencoding/

**Comment by balthild** [comment]
Thanks for reporting! This should be fixed in v0.6.2.

**Comment by 42plamusse** [comment]
Many thanks for your responsiveness !
